File: bharat-railway-backend/src/main/java/com/bharatrailway/auth/application/dto/LoginResponse.java
/*
 * Project: Bharat Railway Booking System
 * Module: bharat-railway-backend
 * Branch: feature/backend-developer-hitanshu
 * Developer: Hitanshu Dhakrey
 * Assisted by: Chandra Shekhar Bansal (Infrastructure), DeepSeek (AI Scribe)
 * Date: 2026-06-20
 * Version: 0.1.0-SNAPSHOT
 *
 * Description:
 * Login response DTO.
 * Returns JWT access token and session UUID on successful authentication.
 */

package com.bharatrailway.auth.application.dto;

import java.util.UUID;

public class LoginResponse {

    private String accessToken;
    private String tokenType;
    private long expiresIn;
    private UUID sessionId;

    public LoginResponse() {
    }

    public LoginResponse(String accessToken, long expiresIn, UUID sessionId) {
        this.accessToken = accessToken;
        this.tokenType = "Bearer";
        this.expiresIn = expiresIn;
        this.sessionId = sessionId;
    }

    public String getAccessToken() {
        return accessToken;
    }

    public void setAccessToken(String accessToken) {
        this.accessToken = accessToken;
    }

    public String getTokenType() {
        return tokenType;
    }

    public long getExpiresIn() {
        return expiresIn;
    }

    public UUID getSessionId() {
        return sessionId;
    }
}

File: bharat-railway-frontend/lib/models/responses/login_response.dart
/*
 * Project: Bharat Railway Booking System
 * Module: bharat-railway-frontend
 * Branch: feature/frontend-developer-chandrashekhar
 * Developer: Chandra Shekhar Bansal
 * Assisted by:
 * Date: 2026-07-06
 * Version: 1.0.0
 *
 * Description:
 * Login response DTO for the authentication API.
 * Maps to backend LoginResponse.java.
 * Contains accessToken, tokenType, expiresIn, and sessionId.
 * Used after successful POST /api/v1/auth/login.
 */

class LoginResponse {
  final String accessToken;
  final String tokenType;
  final int expiresIn;
  final String sessionId;

  LoginResponse({
    required this.accessToken,
    required this.tokenType,
    required this.expiresIn,
    required this.sessionId,
  });

  factory LoginResponse.fromJson(Map<String, dynamic> json) {
    return LoginResponse(
      accessToken: json['accessToken'] as String,
      tokenType: json['tokenType'] as String,
      expiresIn: json['expiresIn'] as int,
      sessionId: json['sessionId'] as String,
    );
  }
}